### ISSUE TITLE
Implement RFC0052: Graceful Stack Upgrades

### ISSUE THREAD

**Issue by ryanmoran** [paketo-buildpacks/php#572]
[RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0052-graceful-stack-upgrades.md)

## Summary

All Paketo Buildpacks should support graceful stack upgrades between the current Bionic and Jammy stacks. For most buildpacks, this should not require any changes, but for some buildpacks, layers created in previous builds that would normally be reused should be recreated if the stack ID changes.

## Action Item
Consider what types of buildpacks would be impacted by a stack upgrade and make any relevant changes that would help to mitigate issues caused by that upgrade. If there is no work to be done, please comment on this issue explaining the context that helped to arrive at that conclusion.

**Comment by fg-j** [comment]
Blocked by https://github.com/paketo-buildpacks/php/issues/558

**Comment by sophiewigmore** [comment]
In investigating this issue, I didn't encounter any issues with stack upgrades for any of the [PHP sample apps](https://github.com/paketo-buildpacks/samples/tree/main/php) (except the vendored app, but that's a separate issue). I also built a laravel app with the steps outlined [here](https://github.com/paketo-buildpacks/composer-install/issues/46#issue-1357771122). That app fails to build due to issues with Jammy, unrelated to upgrading the stack.

Even though I have not seen any failures, I think it would be prudent to clear the composer-install buildpack composer packages cache on stack upgrades. Some packages rely on dependencies like openssl and I think there is a chance that reusing the packages could cause issues with extensions.

**Comment by sophiewigmore** [comment]
https://github.com/paketo-buildpacks/php/releases/tag/v2.0.0
